docs/tutorial/durham_reframe_tutorial.md

```diff
@@ -0,0 +1,409 @@
+# Automating benchmarks with ReFrame
+
+## Outline
+
+1. How ReFrame executes tests
+2. Structure of a ReFrame test -- Hello world example
+3. Configuring ReFrame to run tests on Cosma
+4. Writing performance tests -- Stream example
+5. Working with build systems -- Make, CMake, Autotools, Spack examples
+6. Avoiding build systems -- Run-only tests
+
+Adapted from [ReFrame 4.5 tutorials](https://reframe-hpc.readthedocs.io/en/v4.5.2/tutorials.html) that cover more features.
+There's a [new tutorial](https://reframe-hpc.readthedocs.io/en/v4.6.0/tutorial.html) in ReFrame 4.6
+
+---
+
+## How ReFrame executes tests
+
+When ReFrame executes a test it runs a pipeline of the following stages
+
+![](https://reframe-hpc.readthedocs.io/en/stable/_images/pipeline.svg)
+
+You can customise the behaviour of each stage or add a hook before or after each of them.  For more details, read the [ReFrame pipeline documentation](https://reframe-hpc.readthedocs.io/en/stable/pipeline.html).
+
+---
+
+## Set up environment
+
+This tutorial was originally run on the [Cosma](https://cosma.readthedocs.io/en/latest/) supercomputer. It should be straightforward to run on a different platform, the requirements are  `gcc`, `git` and `python3`. (for the later parts you also need `make`, `autotools`, `cmake` and `spack`).
+Before proceeding to install ReFrame, we recommend creating a python virtual environment to avoid clashes with other installed python packages. First load a newer python module.
+
+```bash
+module swap python/3.10.12
+```
+
+Then create an environment and activate it with
+
+```bash
+python3 -m venv reframe_tutorial
+source reframe_tutorial/bin/activate
+```
+
+You will have to activate the environment each time you login. To deactivate the environment run `deactivate`.
+
+----
+
+## Install ReFrame
+
+Then install ReFrame with `pip`. I am installing version `4.5.2` because we will follow tutorials that have been changed in the latest `4.6.0` version.
+
+```bash
+pip install reframe-hpc==4.5.2
+```
+
+Alternatively, you can
```

Comment:
Say something here like cloning the ReFrame repo if you want access to the tutorials - paths are referenced below.

Reply:
You also get the tutorials if you pip install. But yeah maybe worth saying the tutorials are in there.